tests/test_packages/test_connections/test_http/test_http.py

```diff
@@ -0,0 +1,347 @@
+from aea.mail.base import Envelope
+
+from packages.fetchai.connections.http.connection import HTTPConnection
+
+from ....conftest import ROOT_DIR, get_unused_tcp_port
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+class TestHTTPConnectionConnectDisconnect:
+    """Test the packages/fetchai/connection/http/connection.py."""
+
+    @classmethod
+    def setup_class(cls):
+        """Initialise the class and test connect."""
+
+        cls.address = "my_key"
+        cls.host = "127.0.0.1"
+        cls.port = get_unused_tcp_port()
+        cls.api_spec_path = os.path.join(ROOT_DIR, "tests", "data", "petstore_sim.yaml")
+        cls.connection_id = PublicId("fetchai", "http", "0.1.0")
+        cls.protocol_id = PublicId("fetchai", "http", "0.1.0")
+
+        cls.http_connection = HTTPConnection(
+            address=cls.address,
+            host=cls.host,
+            port=cls.port,
+            api_spec_path=cls.api_spec_path,
+            connection_id=cls.connection_id,
+            restricted_to_protocols=set([cls.protocol_id]),
+        )
+        assert cls.http_connection.channel.is_stopped
+
+        cls.http_connection.channel.connect()
+        assert not cls.http_connection.channel.is_stopped
+
+    @pytest.mark.asyncio
+    async def test_http_connection_disconnect_channel(self):
+        """Test the disconnect."""
+        self.http_connection.channel.disconnect()
+        assert self.http_connection.channel.is_stopped
+
+
+@pytest.mark.asyncio
+class TestHTTPConnectionSend:
+    """Test the packages/fetchai/connection/http/connection.py."""
+
+    @classmethod
+    def setup_class(cls):
+        """Initialise the class."""
+
+        cls.address = "my_key"
+        cls.host = "127.0.0.1"
+        cls.port = get_unused_tcp_port()
+        cls.api_spec_path = os.path.join(ROOT_DIR, "tests", "data", "petstore_sim.yaml")
+        cls.connection_id = PublicId("fetchai", "http", "0.1.0")
+        cls.protocol_id = PublicId("fetchai", "http", "0.1.0")
+
+        cls.http_connection = HTTPConnection(
+            address=cls.address,
+            host=cls.host,
+            port=cls.port,
+            api_spec_path=cls.api_spec_path,
+            connection_id=cls.connection_id,
+            restricted_to_protocols=set([cls.protocol_id]),
+        )
+        loop = asyncio.get_event_loop()
+        value = loop.run_until_complete(cls.http_connection.connect())
+        assert value is None
+        assert cls.http_connection.connection_status.is_connected
+
+    @pytest.mark.asyncio
+    async def test_send_connection(self):
+        """Test send connection error."""
+        client_id = "to_key"
+        envelope = Envelope(
+            to=client_id,
+            sender="from_key",
+            protocol_id=self.protocol_id,
+            message=b"some message",
+        )
+        await self.http_connection.send(envelope)
+        assert (
+            self.http_connection.channel.dispatch_ready_envelopes.get(client_id)
```

Comment:
If the above comment is relevant, this is not ideal. We should put a "dummy request-id" such that we "fool" the `send` method making him believe that there's a pending request.
https://github.com/fetchai/agents-aea/pull/895/files#r388648114

Reply:
done